packages/connect-disputable-voting/src/models/ArbitratorFee.ts

```diff
@@ -0,0 +1,40 @@
+import { SubscriptionHandler } from '@aragon/connect-types'
+
+import ERC20 from './ERC20'
+import { formatBn } from '../helpers'
+import { ArbitratorFeeData, IDisputableVotingConnector } from '../types'
+
+export default class ArbitratorFee {
+  #connector: IDisputableVotingConnector
+
+  readonly id: string
+  readonly voteId: string
+  readonly tokenId: string
+  readonly tokenDecimals: string
+  readonly amount: string
+
+  constructor(
+    data: ArbitratorFeeData,
+    connector: IDisputableVotingConnector
+  ) {
+    this.#connector = connector
+
+    this.id = data.id
+    this.voteId = data.voteId
+    this.tokenId = data.tokenId
+    this.tokenDecimals = data.tokenDecimals
+    this.amount = data.amount
+  }
+
+  get formattedAmount(): string {
+    return formatBn(this.amount, this.tokenDecimals)
+  }
+
+  async token(): Promise<ERC20> {
+    return this.#connector.ERC20(this.tokenId)
+  }
+
+  onToken(callback: Function): SubscriptionHandler {
+    return this.#connector.onERC20(this.tokenId, callback)
+  }
```

Comment:
If we want this method to be supported by `createAppHook()` in the React API, we should do this instead:

```suggestion
  onToken(callback?: SubscriptionCallback<ERC20>): SubscriptionResult<ERC20> {
    return subscription<ERC20>(callback, (callback) =>
      this.#connector.onERC20(this.tokenId, callback)
    )
  }
```

You can find an example here: https://github.com/aragon/connect/blob/3bef799a65a985fbd63949f7fe3f5b852130d5e1/packages/connect-tokens/src/models/Tokens.ts#L37-L44